src/Control/Monad/Trans/Instances.hs

```diff
@@ -1248,3 +1252,129 @@ data C1_0'SelectT
 #  endif
 # endif
 #endif
+
+#if !(MIN_VERSION_base(4,9,0))
+#if MIN_VERSION_transformers(0,4,0) && !(MIN_VERSION_transformers(0,5,0))
+
+#if MIN_VERSION_base(4,4,0)
```

Comment:
Style nitpick: to make it easier to tell at what level of nested CPP we're in, I prefer to add a number of spaces equal to the level. Since this is two levels of CPP, I would indent it as:

```suggestion
#  if MIN_VERSION_base(4,4,0)
```

And similarly for other `#if`s and `#endif`s in this block of code.